FuelLabs/fuel-indexer
# enhancement: add `forc index revert` command

Closes #464

## Changelog 
• creates new `forc index revert command` and associated files for the new command to work.  
• updates docs for the new command
• adds new route for the revert op
• adds new queries method `penultimate_index_id_for`
• adds new match arm for `create_service_task()` inside `service.rs`.

## Testing
1. Start the service with a fresh database and using the block explorer example: `cargo run --bin fuel-indexer -- run --fuel-node-host node-beta-2.fuel.network --fuel-node-port 80`
2. Deploy the block explorer example with a log line.
3. Deploy the block explorer example a second time with a different log line.
4. Stop the service.
5. Immediately start it again.
6. Revert the index WASM using the explorer manifest.

 ```
 ⬅️  Revertingindex 'my_project.test_index' at http://127.0.0.1:29987/api/index/my_project/test_index

{
  "success": "true"
}

✅ Successfully reverted index 'my_project.test_index' at http://127.0.0.1:29987/api/index/my_project/test_index
```
If you don't see this make sure to rebuild the binary in step 1.

7. Trigger some events manually 
8. Check in the DB that the penultimate id of `hello-index` (id 1) is indexing the events, not id 2

## Notes
I noticed that in all the other plugin commands, the service is interacted with via the api, which I think is a good and clean approach. As there was no api route for reverting an index, I created a new one. This then triggers the `create_servce_task()` and the new `ServiceRequest::IndexRevert` is matched and executed.
